packages/flutter/lib/src/painting/continuous_stadium_border.dart

```diff
@@ -0,0 +1,278 @@
+/// }
+/// ```
+/// {@end-tool}
+///
+/// See also:
+///
+/// * [RoundedRectangleBorder] which creates a rectangle whose corners are
+///   precisely quarter circles.
+/// * [ContinuousRectangleBorder] which creates a rectangle whose 4 edges have
+///   a continuous transition into each of its four corners.
+class ContinuousStadiumBorder extends ShapeBorder {
+  /// Creates a Continuous Stadium Border.
+  ///
+  /// The [side], and [borderRadius] arguments must not be null.
+  const ContinuousStadiumBorder({
+    this.side = BorderSide.none,
+  }) : assert(side != null);
+
+  /// The style of this border.
+  ///
+  /// By default this value is [BorderSide.none]. It also must not be null.
+  final BorderSide side;
+
+  Path _getPath(Rect rect) {
+    // The radius multiplier where the resulting shape will perfectly concave at
+    // with a height and width of any value.
+    const double maxMultiplier = 3.0573;
+
+    // The multiplier of the radius in comparison to the smallest edge length
+    // used to describe the minimum radius for this shape.
+    const double dynamicShapeMinMultiplier = 1 / maxMultiplier;
+
+    // The maximum aspect ratio of the width and height of the given rect before
+    // clamping on one dimension will occur.
+    const double maxAspectRatio = 1 - dynamicShapeMinMultiplier;
+
+    final double rectWidth = rect.width;
+    final double rectHeight = rect.height;
+    final bool widthLessThanHeight = rectWidth < rectHeight;
+    final double width = widthLessThanHeight ? rectWidth.clamp(0.0, maxAspectRatio * rectHeight) : rectWidth;
+    final double height = widthLessThanHeight ? rectHeight : rectHeight.clamp(0.0, maxAspectRatio * rectWidth);
+    final double centerX = rect.center.dx;
+    final double centerY = rect.center.dy;
+    final double originX = centerX - width / 2;
+    final double originY = centerY - height / 2;
+    final double minDimension = math.min(width, height);
+    final double radius = minDimension * dynamicShapeMinMultiplier;
+    final double limitedRadius = math.min(radius, minDimension * dynamicShapeMinMultiplier);
+
+    // These equations give the x and y values for each of the 8 mid and corner
+    // points on a rectangle.
+    double leftX(double x) { return centerX + x * limitedRadius - width / 2; }
+    double rightX(double x) { return centerX - x * limitedRadius + width / 2; }
+    double topY(double y) { return centerY + y * limitedRadius - height / 2; }
+    double bottomY(double y) { return centerY - y * limitedRadius + height / 2; }
+    double bottomMidY(double y) { return originY + height - y * limitedRadius; }
+    double leftMidX(double x) { return originX + x * height; }
+    double rightMidX(double x) { return originX + width - x * limitedRadius; }
+
+    // The secondary super elliptical shape where there are only 2 straight edges
+    // and two 180º curves. The width is greater than the height. Approximately
+    // renders a super ellipse with an n value of 2.
```

Comment:
A superellipse with n=2 is just an ellipse no?

Reply:
yup